lua/lazyvim/plugins/extras/lang/rust.lua

```diff
@@ -0,0 +1,132 @@
+return {
+
+  -- Extend auto completion
+  {
+    "hrsh7th/nvim-cmp",
+    dependencies = {
+      {
+        "Saecki/crates.nvim",
+        event = { "BufRead Cargo.toml" },
+        config = true,
+      },
+    },
+    ---@param opts cmp.ConfigSchema
+    opts = function(_, opts)
+      local cmp = require("cmp")
+      opts.sources = cmp.config.sources(vim.list_extend(opts.sources, {
+        { name = "crates" },
+      }))
+    end,
+  },
+
+  -- Add Rust & related to treesitter
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "ron", "rust", "toml" })
+      end
+    end,
+  },
+
+  -- Ensure Rust debugger is installed
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "codelldb" })
+      end
+    end,
+  },
+
+  -- Correctly setup lspconfig for Rust 🚀
+  {
+    "neovim/nvim-lspconfig",
+    dependencies = { "simrat39/rust-tools.nvim" },
+    opts = {
+      servers = {
+        -- Ensure mason installs the server
+        rust_analyzer = {},
+        taplo = {},
+      },
+      setup = {
+        rust_analyzer = function(_, opts)
+          require("lazyvim.util").on_attach(function(client, buffer)
+						-- stylua: ignore
+						if client.name == "rust_analyzer" then
+							vim.keymap.set("n", "K", "<cmd>RustHoverActions<cr>", { buffer = buffer, desc = "Hover Actions (Rust)" })
+							vim.keymap.set("n", "<leader>cR", "<cmd>RustCodeAction<cr>", { buffer = buffer, desc = "Code Action (Rust)" })
+			  			vim.keymap.set("n", "<leader>dr", "<cmd>RustDebuggables<cr>", { buffer = buffer, desc = "Run Debuggables (Rust)" })
+						end
+          end)
+          local mason_registry = require("mason-registry")
+          -- rust tools configuration for debugging support
+          local codelldb = mason_registry.get_package("codelldb")
+          local extension_path = codelldb:get_install_path() .. "/extension/"
+          local codelldb_path = extension_path .. "adapter/codelldb"
+          local liblldb_path = vim.fn.has("mac") == 1 and extension_path .. "lldb/lib/liblldb.dylib"
+            or extension_path .. "lldb/lib/liblldb.so"
+          local rust_tools_opts = vim.tbl_deep_extend("force", opts, {
+            dap = {
+              adapter = require("rust-tools.dap").get_codelldb_adapter(codelldb_path, liblldb_path),
+            },
```

Comment:
Pretty novice with Lua here, and I had trouble figuring out how to successfully run with a fork/branch of LazyVim, in order to test my own attempts at addressing the concern below. Pointers welcome!

This code has an implicit hard requirement on `mason`, which I keep disabled because I manage CLIs separately. Any chance this module's implementation could be refactored slightly to fail gracefully is the `mason-registry` require fails and still set up any of the options that can be determined without it? It looks like it would just be the DAP piece that depends on it?

Reply:
I refactored the whole rust extra. It can now work without mason and you can easily override the rust-tools options.